File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic VB Programming/Inheritance/Tester.vb
Module Tester

    Sub Main()
        Dim e1 As New Employee("ABC123", "Joe Bloggs", 27, 20000)
        Dim e2 As New SalesPerson("DEF456", "Jane Smith", 32, 40000, 120000)

        Console.WriteLine(vbNewLine + "------Before awarding bonus------" + vbNewLine)
        Console.WriteLine("e1 is {0}", e1)
        Console.WriteLine("e2 is {0}", e2)

        e1.awardBonus(500)
        e2.awardBonus(500)

        Console.WriteLine(vbNewLine + "------After awarding bonus------" + vbNewLine)
        Console.WriteLine("e1 is {0}", e1)
        Console.WriteLine("e2 is {0}", e2)

    End Sub

End Module

File: CourseProjects/CourseProjects/VisualStudioProjects/Languages/Basic CS  Programming/Inheritance/Tester.cs
using System;

namespace InheritanceDemo
{
	class Tester {

		static void Main(string[] args) {
			Employee e1 = new Employee("ABC123","Joe Bloggs",27,20000);
			Employee e2 = new SalesPerson("DEF456","Jane Smith",32,40000,120000);

			Console.WriteLine("\n------Before awarding bonus------");
			Console.WriteLine("e1 is {0}\ne2 is {1}",e1,e2);

			e1.awardBonus(500);
			e2.awardBonus(500);

			Console.WriteLine("\n------After awarding bonus------");
			Console.WriteLine("e1 is {0}\ne2 is {1}",e1,e2);

		}
	}
}
